### PR TITLE
fix(): update gw deploy if gateway sidecar image changed

### PR DESCRIPTION
**Pull request by mridulgain**
# Description

Fixes: update if gateway sidecar image has been changed in worker-operator env vars

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have ran `go fmt`
* [ ] I have updated the helm chart as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit test cases.
* [ ] I have verified the E2E test cases with new code changes.
* [ ] I have added all the required E2E test cases.

## Does this PR introduce a breaking change?

```release-note

```

### CODE REVIEW

**Review by narmidm — APPROVED** [review]

**Comment by narmidm** [comment]
LGTM! Any test case changes required?

**Comment by narmidm** [comment]
LGTM! Any test case changes required?

**Comment by mridulgain** [comment]
> LGTM! Any test case changes required?

No. But it needed an empty string check. Thanks!

File: controllers/slicegateway/slicegateway.go [hunk 1]
```diff
@@ -1386,6 +1395,27 @@ func (r *SliceGwReconciler) ReconcileGatewayDeployments(ctx context.Context, sli
 				return ctrl.Result{}, err, true
 			}
 			return ctrl.Result{Requeue: true}, nil, true
+		} else {
+			// update logic for gateways
+			for i := range deployments.Items {
```
Comment (bharath-avesha) [inline controllers/slicegateway/slicegateway.go:1400]: The "if" conditional above goes over the list of deployments indirectly through the numGwInstances, so it is not appropriate to loop over the deployments afresh in the "else" block. Please move the image check outside the numGwInstances "for" loop. Create a new loop that goes over each deployment and update if there is an image mismatch.

**Pull request merged**